src/modules/quick_api_module.f90

```diff
@@ -0,0 +1,688 @@
+!---------------------------------------------------------------------!
+! Created by Madu Manathunga on 04/16/2020                            !
+!                                                                     ! 
+! Copyright (C) 2020-2021 Merz lab                                    !
+! Copyright (C) 2020-2021 Götz lab                                    !
+!                                                                     !
+! This Source Code Form is subject to the terms of the Mozilla Public !
+! License, v. 2.0. If a copy of the MPL was not distributed with this !
+! file, You can obtain one at http://mozilla.org/MPL/2.0/.            !
+!_____________________________________________________________________!
+
+! Interface for quick libarary
+module quick_api_module
+
+  implicit none
+  private
+
+  public :: quick_api
+  public :: setQuickJob, getQuickEnergy, getQuickEnergyGradients, deleteQuickJob
+
+#ifdef MPIV
+  public :: setQuickMPI
+#endif
+
+  type quick_api_type
+    ! indicates if quick should run in library mode. This will help
+    ! setting up files for quick run. 
+    logical :: apiMode = .false.
+
+    ! used to determine if memory should be allocated and SAD guess should
+    ! be performed
+    logical :: firstStep = .true.
+
+    ! current md step, should come from MM code
+    integer :: mdstep = 1
+
+    ! keeps track of how many times quick is called by MM code 
+    integer :: step = 1
+
+    ! number of atoms
+    integer :: natoms = 0
+
+    ! number of atom types
+    integer :: natm_type = 0
+
+    ! number of external point charges
+    integer :: nxt_ptchg = 0
+
+    ! atom type/atomic number of each atom
+    integer, allocatable, dimension(:) :: atm_type_id
+
+    ! atomic numbers of atoms
+    integer, allocatable, dimension(:) :: atomic_numbers
+
+    ! xyz coordinates of atoms, size is 3*natoms
+    double precision, allocatable, dimension(:,:) :: coords
+
+    ! charge and the coordinates of external point charges
+    ! size is 4*nxt_ptchg, where first element of a row holds the charge
+    double precision, allocatable, dimension(:,:) :: xt_chg_crd
```

Comment:
Maybe be consistent in naming. Here you use xt_chg, below ptchg

Reply:
Fixed.